lib/Importer/ProtobufLoader.cpp

```diff
@@ -101,16 +100,34 @@ bool ProtobufLoader::hasNodeByName(llvm::StringRef name) const {
 }
 
 ProtobufLoader::ProtobufLoader(llvm::ArrayRef<const char *> tensorNames,
-                               llvm::ArrayRef<TypeRef> types, Function &F)
+                               llvm::ArrayRef<TypeRef> types, Function &F,
+                               llvm::Error *errPtr)
     : G_(F) {
   // Verify that the version of the library that we linked against is
   // compatible with the version of the headers we compiled against.
   GOOGLE_PROTOBUF_VERIFY_VERSION;
 
-  assert(tensorNames.size() == types.size() && "Invalid initialization list");
-  for (unsigned i = 0; i < tensorNames.size(); i++) {
-    assert(!hasNodeByName(tensorNames[i]) && "Input names have duplicate");
-    TEMP_EXIT_ON_ERR(createAndRegisterPlaceholder(tensorNames[i], types[i]));
+  // Lambda to setup the ProtobufLoader and return any llvm::Errors that were
```

Comment:
Why do we need the Lambda here?

Reply:
This is here because we can't return an error from the constructor but we want to be able to isolate the fallible code in the constructor so that we can 1. reuse the same error handling mechanisms the rest of the loader uses such as `RETURN_IF_ERR`, `ASSIGN_VALUE_OR_RETURN_ERR`, etc and 2. not repeatedly check if errPtr==nullptr and handle each errors independently (we can just say here's the fallible logic in this lambda, if errPtr !=nullptr then assign any error that occurs there otherwise if errPtr==nullptr handle any errors by exiting). I chose to use a lambda and not a private method because this code should not get called outside of this constructor and if anyone wants to create another constructor in the future they can follow this pattern without any issues.